libp2p/protocols/pubsub/pubsubpeer.nim

```diff
@@ -272,9 +272,30 @@ proc sendEncoded*(p: PubSubPeer, msg: seq[byte]) {.raises: [], async.} =
 
     await conn.close() # This will clean up the send connection
 
-proc send*(p: PubSubPeer, msg: RPCMsg, anonymize: bool) {.raises: [].} =
-  trace "sending msg to peer", peer = p, rpcMsg = shortLog(msg)
+iterator splitRPCMsg(peer: PubSubPeer, rpcMsg: RPCMsg, maxSize: int, anonymize: bool): seq[byte] =
+  var currentRPCMsg = RPCMsg(subscriptions: rpcMsg.subscriptions, control: rpcMsg.control,
```

Comment:
This code (& the len part) seems very prone to failure, if we add a field to RPCMsg, update a field, etc
We should at the very least add a comment to the RPCMsg declaration to warn about that, and ideally find a better solution

It seems that the best solution would be to use nim-protobuf-ser, but we can't yet unfortunately

As an alternative, wouldn't it be simpler to modify `encodeRpcMsg`? It could take a `maxSize`, and return a `seq[seq[byte]]`, or even just concatenate the PBs directly internally (would require some tweaks to the pubsubpeer)

Reply:
I think adding the comment is a good and easy-to-do improvement. But not sure about splitting in `encodeRpcMsg`, it seems too much abstraction leakage.

Reply:
Added the comment